src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -954,9 +943,18 @@ def validate_resource_group_parameter(namespace):
         raise RequiredArgumentMissingError("Please specify --resource-group")
 
 
-def validate_location_cluster_parameters_present(namespace):
+def validate_location_resource_group_cluster_parameters(namespace):
+    location = namespace.location
+    resource_group_name = namespace.resource_group_name
+    cluster_name = namespace.cluster_name
     """Validates that either location or cluster details (i.e. resource group and cluster name) are specified"""
-    if not namespace.location and not namespace.cluster_name and not namespace.resource_group_name:
+    if not location and not cluster_name and not resource_group_name:
         raise RequiredArgumentMissingError(
-            "You must specify the location or cluster details (i.e. resource group name and cluster name)."
+            "You must specify --location or --resource-group and --cluster."
+        )
+
+    """Validates that location, cluster name, and resource group name are not specified at the same time"""
+    if location and resource_group_name and cluster_name:
```

Comment:
```suggestion
    if location and (resource_group_name or cluster_name):
```

So you reject `--location` + `--cluster-name` too, which shouldn't be allowed.